### PR TITLE
crimson: use std::strong_ordering mechinary

### PR DESCRIPTION
**Pull request by tchaikov**
## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
this change is a subset of #47427. it is extracted out of #47427 in hope to get reviewed and merged sooner.

File: src/crimson/os/seastore/seastore_types.h [hunk 1]
```diff
@@ -1593,8 +1585,9 @@ struct record_size_t {
   }
 
   void account(const delta_info_t& delta);
+
+  auto operator<=>(const record_size_t &) const = default;
```
Comment (cyx1231st) [inline src/crimson/os/seastore/seastore_types.h:1589]: It doesn't make sense to implement comparison operators here.

== operator is enough.

File: src/crimson/os/seastore/backref/backref_tree_node.h [hunk 2]
```diff
@@ -26,11 +26,11 @@ struct backref_map_val_t {
     laddr_t laddr,
     extent_types_t type)
     : len(len), laddr(laddr), type(type) {}
+
+  bool operator==(const backref_map_val_t& rhs) const noexcept {
+    return len == rhs.len && laddr == rhs.laddr;
```
Comment (cyx1231st) [inline src/crimson/os/seastore/backref/backref_tree_node.h:31]: Seems we need to consider type in the comparison. @xxhdx1985126 is it correct?

Anyway, LGTM in this PR.

File: src/crimson/os/seastore/seastore_types.h (hunk 1 above)
Comment (tchaikov) [reply src/crimson/os/seastore/seastore_types.h]: @cyx1231st fixed and repushed

File: src/crimson/os/seastore/backref/backref_tree_node.h (hunk 2 above)
Comment (tchaikov) [reply src/crimson/os/seastore/backref/backref_tree_node.h:31]: yeah, let's leave it to another PR.

File: src/crimson/os/seastore/seastore_types.h [hunk 3]
```diff
@@ -1739,8 +1732,9 @@ struct record_group_size_t {
 
   void account(const record_size_t& rsize,
                extent_len_t block_size);
+
+  auto operator<=>(const record_group_size_t &) const = default;
```
Comment (cyx1231st) [inline src/crimson/os/seastore/seastore_types.h:1736]: Ah, forget to point out that this is also == operator only.

File: src/crimson/os/seastore/lba_manager/btree/lba_btree_node.h [hunk 4]
```diff
@@ -44,13 +44,8 @@ struct lba_map_val_t {
     uint32_t refcount,
     uint32_t checksum)
     : len(len), paddr(paddr), refcount(refcount), checksum(checksum) {}
+  bool operator==(const lba_map_val_t&) const = default;
```
Comment (cyx1231st) [inline src/crimson/os/seastore/lba_manager/btree/lba_btree_node.h:47]: Commit title: s/us/use

File: src/crimson/os/seastore/seastore_types.h (hunk 3 above)
Comment (tchaikov) [reply src/crimson/os/seastore/seastore_types.h]: fixed and repushed.

File: src/crimson/os/seastore/lba_manager/btree/lba_btree_node.h (hunk 4 above)
Comment (tchaikov) [reply src/crimson/os/seastore/lba_manager/btree/lba_btree_node.h:47]: ack.

**Review by cyx1231st — APPROVED** [review]

**Pull request merged**